array_test.go

```diff
@@ -1290,3 +1290,286 @@ func TestArray_NotFind(t *testing.T) {
 		array.chain.assertFailed(t)
 	})
 }
+func TestArray_IsOrdered(t *testing.T) {
+	type args struct {
+		values      []interface{}
+		less        []func(x, y *Value) bool
+		chainFailed bool
+	}
+	tests := []struct {
+		name   string
+		args   args
+		wantOK bool
+	}{
+		{
+			name: "array boolean ordered",
+			args: args{
+				values: []interface{}{false, false, true, true},
+			},
+			wantOK: true,
+		},
+		{
+			name: "array number ordered",
+			args: args{
+				values: []interface{}{1, 1, 2, 3},
+			},
+			wantOK: true,
+		},
+		{
+			name: "array string ordered",
+			args: args{
+				values: []interface{}{"", "a", "b", "ba"},
+			},
+			wantOK: true,
+		},
+		{
+			name: "array of nil elements",
+			args: args{
+				values: []interface{}{nil, nil, nil},
+			},
+			wantOK: true,
+		},
+		{
+			name: "wrong order",
+			args: args{
+				values: []interface{}{3, 2, 1},
+			},
+			wantOK: false,
+		},
+		{
+			name: "user-defined less function",
+			args: args{
+				values: []interface{}{1, 2, 3},
+				less: []func(x, y *Value) bool{
+					func(x, y *Value) bool {
+						valX := x.Number().Raw()
+						valY := y.Number().Raw()
+						return valX < valY
+					},
+				},
+			},
+			wantOK: true,
+		},
+		{
+			name: "invalid - failed type assertion on less function",
+			args: args{
+				values: []interface{}{1, 2, 3},
+				less: []func(x, y *Value) bool{
+					func(x, y *Value) bool {
+						valX := x.String().Raw()
+						valY := y.String().Raw()
+						return valX < valY
```

Comment:
Let's replace this with return true. This way we can be sure that the failure is caused by assertion, not by return value. So the function will look Iike:


x.String()
y.String()
return true

Reply:
Yes, you're right